sklearn/impute/tests/test_common.py

```diff
@@ -181,3 +181,39 @@ def test_keep_empty_features(imputer, keep_empty_features):
             assert X_imputed.shape == X.shape
         else:
             assert X_imputed.shape == (X.shape[0], X.shape[1] - 1)
+
+
+@pytest.mark.parametrize("imputer", imputers(), ids=lambda x: x.__class__.__name__)
+@pytest.mark.parametrize("missing_value_test", [np.nan, 1])
+def test_imputation_adds_missing_indicator_if_add_indicator_is_true(
+    imputer, missing_value_test
+):
+    """Check that missing indicator always exists when add_indicator=True.
+
+    Non-regression test for gh-26590.
+    """
+    X_train = np.array([[0, np.NaN], [1, 2]])
+
+    # Test data without missing values that can be set to np.NaN or 1.
```

Comment:
This statement seems a bit confusing because when we have np.nan I think it implies that there is a missing value.

Reply:
Is refrasing the comment as "Test data where missing_value_test variable can be explicitly set to either np.NaN or 1" helpful?
```suggestion
    # Test data where missing_value_test variable can be set to np.NaN or 1.
```

Reply:
Yes that looks better.